test/integration/api_test.lua

```diff
@@ -1,12 +1,264 @@
+        datadir = helper.shared.datadir,
+        use_vshard = true,
+        replicasets = {
+            {
+                alias = 'router',
+                uuid = helper.shared.uuid('a'),
+                roles = {
+                    'api',
+                },
+                servers = {
+                    { instance_uuid = helper.shared.uuid('a', 1), alias = 'router' },
+                },
+            },
+            {
+                alias = 'kv-storage',
+                uuid = helper.shared.uuid('b'),
+                roles = {
+                    'kv-storage',
+                },
+                servers = {
+                    { instance_uuid = helper.shared.uuid('b', 1), alias = 'kv-storage-1-master' },
+                    { instance_uuid = helper.shared.uuid('b', 2), alias = 'kv-storage-1-replica-1' },
+                    { instance_uuid = helper.shared.uuid('b', 3), alias = 'kv-storage-1-replica-2' },
+                },
+            },
+        },
+        env = {
+            ['ENGINE'] = 'memtx',
+        },
+    })
+
+    g.cluster:start()
+    g.router = g.cluster.main_server
+end
+
+g.after_all = function()
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+g.before_each(function()
+    for _, server in ipairs(g.cluster.servers) do
+        server.net_box:eval([[
+            local space = box.space.kv
+            if space ~= nil and not box.cfg.read_only then
+                space:truncate()
+            end
+        ]])
+    end
+end)
+
+g.after_each(function()
+
+end)
+
+-- CRUD helpers
+local function create(key, value)
+    return g.router:http_request('post', url..'/', {
+        json = {
+            key = key,
+            value = value,
+        },
+        http = { timeout = 1 },
+        raise = false,
+    })
+end
+
+local function read(key)
+    return g.router:http_request('get', url..'/'..key, { http = { timeout = 1 } , raise = false })
+end
+
+local function update(key, value)
+    local resp = g.router:http_request('put', url..'/'..key, {
+        json = { value = value },
+        http = {
+            timeout = 1
+        },
+        raise = false,
+    })
+    log.info(resp)
+    return resp
+end
+
+local function delete(key)
+    return g.router:http_request('delete', url..'/'..key, { http = { timeout = 1 }, raise = false })
+end
+
+
+-- Is request ok helpers
+local function ok(resp, expected_body)
+    t.assert_equals(resp.status, 200)
+    if expected_body ~= nil then
+        t.assert_equals(resp.json, expected_body)
+    end
+end
+
+local function create_ok(key, value)
+    ok(create(key, value), { result = 'record created' })
+end
+
+local function read_ok(key, expected_value)
+    ok(read(key), { value = expected_value })
+end
+
+local function update_ok(key, value)
+    ok(update(key, value), { result = 'record updated' })
+end
+
+local function delete_ok(key)
+    ok(delete(key), { result = 'record deleted' })
+end
+
```

Comment:
I like this approach, but function names can be better:

*  `ok` -> `assert_resp_ok`
* `create_ok` - > `assert_create_ok`

and so on.